neo/renderer/RenderSystem_init.cpp

```diff
@@ -1261,67 +1299,86 @@ Downsample is the number of steps to mipmap the image before saving it
 If ref == NULL, session->updateScreen will be used
 ==================
 */
-void idRenderSystemLocal::TakeScreenshot( int width, int height, const char *fileName, int blends, renderView_t *ref ) {
-	byte		*buffer;
-	int			i, j, c, temp;
+void idRenderSystemLocal::TakeScreenshot( int width, int height, const char *fileName, int blends, renderView_t *ref, int overrideFormat ) {
+	byte		*buffer, *flippedBuffer;
+	int			i, j, k;
 
 	takingScreenshot = true;
 
 	int	pix = width * height;
 
-	buffer = (byte *)R_StaticAlloc(pix*3 + 18);
-	memset (buffer, 0, 18);
+	buffer = (byte *)R_StaticAlloc(pix*3);
+	flippedBuffer = (byte*)R_StaticAlloc(pix*3);
 
 	if ( blends <= 1 ) {
-		R_ReadTiledPixels( width, height, buffer + 18, ref );
+		R_ReadTiledPixels( width, height, buffer, ref );
 	} else {
 		unsigned short *shortBuffer = (unsigned short *)R_StaticAlloc(pix*2*3);
-		memset (shortBuffer, 0, pix*2*3);
+		memset( shortBuffer, 0, pix*2*3 );
 
 		// enable anti-aliasing jitter
 		r_jitter.SetBool( true );
 
 		for ( i = 0 ; i < blends ; i++ ) {
-			R_ReadTiledPixels( width, height, buffer + 18, ref );
+			R_ReadTiledPixels( width, height, buffer, ref );
 
 			for ( j = 0 ; j < pix*3 ; j++ ) {
-				shortBuffer[j] += buffer[18+j];
+				shortBuffer[j] += buffer[j];
 			}
 		}
 
 		// divide back to bytes
 		for ( i = 0 ; i < pix*3 ; i++ ) {
-			buffer[18+i] = shortBuffer[i] / blends;
+			buffer[i] = shortBuffer[i] / blends;
 		}
 
 		R_StaticFree( shortBuffer );
 		r_jitter.SetBool( false );
 	}
 
-	// fill in the header (this is vertically flipped, which qglReadPixels emits)
-	buffer[2] = 2;		// uncompressed type
-	buffer[12] = width & 255;
-	buffer[13] = width >> 8;
-	buffer[14] = height & 255;
-	buffer[15] = height >> 8;
-	buffer[16] = 24;	// pixel size
+	// The buffer is upside down, we need to flip it the right way.
+	for (i = 0; i < width; i++) {
+		for (j = 0; j < height; j++) {
+			for (k = 0; k < 3; k++) {
+				flippedBuffer[(i + j * width) * 3 + k] = buffer[(i + (height - 1 - j) * width) * 3 + k];
```

Comment:
I don't think you need a flipped buffer and all this code.
just create a small buffer that can hold one line of pixels, and then use memcpy() to swap the buffer line-wise inplace.
somehow like this (untested): 
```c++
int lineSize = width*3;
byte* swapBuf = _alloca(lineSize);

for(int i=0; i < height/2; ++i) {
	byte* line1 = &buffer[i*lineSize];
	byte* line2 = &buffer[(height-i-1)*lineSize];
	memcpy(swapBuf, line1, lineSize);
	memcpy(line1, line2, lineSize);
	memcpy(line2, swapBuf, lineSize);
}
```

Reply:
This works. Adding! 🎉